packages/wdio-browserstack-service/src/testOps/featureUsage.ts

```diff
@@ -0,0 +1,70 @@
+class FeatureUsage {
+    private isTriggered?: boolean
+    private status?: string
+    private error?: string
+
+    constructor(isTriggered?: boolean) {
+        if (isTriggered !== undefined) {
+            this.isTriggered = isTriggered
+        }
+    }
+
+    public getTriggered(): boolean | undefined {
+        return this.isTriggered
+    }
+
+    public setTriggered(triggered: boolean): void {
+        this.isTriggered = triggered
+    }
+
+    public setStatus(status: string): void {
+        this.status = status
+    }
+
+    public setError(error: string): void {
+        this.error = error
+    }
+
+    public triggered(): void {
+        this.isTriggered = true
+    }
+
+    public failed(e: unknown): void {
+        this.status = 'failed'
+        if (typeof e === 'string') {
```

Comment:
is this a valid case or just a fall back for retrieving stacktrace ?

Reply:
Yes valid case as used over here

https://github.com/sriteja777/webdriverio/blob/ddf474a41efda17e608b7044da9381457c9e5aa5/packages/wdio-browserstack-service/src/util.ts#L558